src/ufuncs.c

```diff
@@ -818,10 +818,21 @@ static void Y_d_func(char** args, npy_intp* dimensions, npy_intp* steps, void* d
 static PyUFuncGenericFunction Y_d_funcs[1] = {&Y_d_func};
 
 /* Define the geom -> int functions (Y_i) */
-static void* get_type_id_data[1] = {GEOSGeomTypeId_r};
-static void* get_dimensions_data[1] = {GEOSGeom_getDimensions_r};
-static void* get_coordinate_dimension_data[1] = {GEOSGeom_getCoordinateDimension_r};
-static void* get_srid_data[1] = {GEOSGetSRID_r};
+/* data values are GEOS func, GEOS error code, return value when input is None */
+static void* get_type_id_func_tuple[3] = {GEOSGeomTypeId_r, (void*)-1, (void*)-1};
+static void* get_type_id_data[1] = {get_type_id_func_tuple};
```

Comment:
Defining it in two steps like this is needed? 
(the `_data` object needs to have length 1?)

Reply:
Yes, it appeared necessary to do this in 2 parts, like in #92.  I got segfaults when doing it in one step and didn't investigate further once the 2 part approach solved it.

The [numpy ufuncs custom example](https://numpy.org/doc/stable/user/c-info.ufunc-tutorial.html?highlight=pyufuncgenericfunction) includes an example of a 4 element `data` value - which implies it should have worked originally, but the example doesn't include mixed types like here (shouldn't matter though?).

@caspervdw any insight into why #92 and here needed to pass these parameters in 2 parts?

Reply:
Yes I investigated this indeed.

This is actually explained https://numpy.org/doc/stable/reference/c-api/ufunc.html#functions .

> data – Should be NULL or a pointer to an array of size ntypes . 

Where ntypes is:

> ntypes – How many different data-type-specific functions the ufunc has implemented.

In other words, numpy allows for providing multiple functions to let a function accept for example double and single precision floats. The example that @brendan-ward illustrates that nicely - the function there has `data` of length four, but also:

```
PyUFuncGenericFunction funcs[4] = {&half_float_logit,
                                   &float_logit,
                                   &double_logit,
                                   &long_double_logit};
```